### PR TITLE
Removed `SessionIdentifierAwareInterface`

### LINKED ISSUE #59: Deprecate `SessionIdentifierAwareInterface` for removal in v2.0
#### Summary

The method `getId` should be present in `SessionInterface` instead of in the separate interface `SessionIdentifierAwareInterface`.

`SessionIdentifierAwareInterface` needs to be deprecated in 1.x and merged into `SessionInterface` in 2.0

### PR DESCRIPTION
**Pull request by alexmerlin**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | yes
| New Feature   | no
| RFC           | no
| QA            | no

### Description

As requested in #59 this commit:

- adds `SessionIdentifierAwareInterface`'s `getId()` method to `SessionInterface`
- removes `SessionIdentifierAwareInterface`

Also, adds `v2` documentation.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -16,8 +23,9 @@ extra:
     project: Mezzio
     installation:
         config_provider_class: 'Mezzio\Session\ConfigProvider'
-    current_version: v1
+    current_version: v2
     versions:
+      - v2
       - v1
 plugins:
     - redirects:
```
Comment (gsteel) [inline mkdocs.yml:31]: The redirects should be changed to point to the v2 equivalent - when v2 is released, we want old urls to redirect to the most recent version

**Review by gsteel — CHANGES_REQUESTED** [review]
Code changes look 👍 - Just the redirects need updating

File: mkdocs.yml (hunk 1 above)
Comment (alexmerlin) [reply mkdocs.yml:31]: Modified redirects to go to `v2`.
Also, added `non-empty-string` to `SessionInterface::getId()` and adapted code where needed.

**Review by gsteel — CHANGES_REQUESTED** [review]
Sorry @alexmerlin - I've just been checking the implementations due to the removed test `testGetIdReturnsEmptyStringIfNoIdentifierProvidedToConstructor`, and the session id is expected to be `''|non-empty-string`. This is used to signal to persistence implementations that the session needs to be regenerated - My fault. It might have been clearer for the id to be typed as `null|non-empty-string` but it's not worth changing this for all the implementations.

I should have checked this out properly before opening my mouth! So the addition of `non-empty-string` needs to be reverted.

For future reference, the `@psalm-` prefix is generally unnecessary. Most IDEs along with Psalm and PHPStan understand `non-empty-string` and most other annotations. Notable exceptions to this rule are `@psalm-type` and `@psalm-import-type`

**Review by gsteel — APPROVED** [review]
Thanks @alexmerlin 👍

**Comment by froschdesign** [comment]
@alexmerlin @gsteel 
Please make sure that all changes in the history are visible. The changes to the documentation are not available here as a separate pull request/commit.

**Pull request merged**